xamarin/Xamarin.Forms
# [Core] Fix wrong orientation issue using OrientationStateTrigger

### Description of Change ###

Fix wrong orientation issue using `OrientationStateTrigger`.

### Issues Resolved ### 

- fixes #14472 
- fixes #9651

### API Changes ###
 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)


### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###
Launch Core Gallery and navigate to the issue 14472. If can see a Green Background Color in PortraitDown orientation, the test has passed.

### PR Checklist ###
